packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -335,6 +335,11 @@ export function createRouteManifest(
 				const route = `/${segments
 					.map(([{ dynamic, content }]) => (dynamic ? `[${content}]` : content))
 					.join('/')}`.toLowerCase();
+				const locale = settings.config.experimental.i18n?.locales.find((currentLocale) => {
+					if (route.includes(`/${currentLocale}`)) {
+						return currentLocale;
+					}
+				});
```

Comment:
Do we allow the locale directory to be anywhere in the route? e.g. `src/pages/foo/bar/pt/content.astro`. I think it's most common being strictly `src/pages/pt/...` and maybe we should change the check here from `.includes` to `.startsWith`.

And perhaps check with `/${currentLocale}/` instead (two wrapping slashes), so `pt` doesn't accidentally match `/pt-br/...`.

Would be nice to wrap this as a function too.

Reply:
Yes, I believe the plan is that it has to be the route, as this is how it works in other frameworks and in Starlight. We discussed some ways to make it configurable but haven't come up with a good solution as of yet.

Reply:
> .includes to .startsWith

My initial implementation used `startsWith`,  but that wouldn't work in cases where the user decides to use `base` in their configuration. 

> And perhaps check with /${currentLocale}/ instead (two wrapping slashes), so pt doesn't accidentally match /pt-br/....

That wouldn't work in cases we a plain `index` that hasn't been translated.

So if you have `locales: ['en',  'it']` and you **don't have** the page `src/pages/it/index.mdx`, then when you call `example.com/it`, you'd want to show the english version.

Reply:
> We can get the base from settings.config.base here.

Yeah I know, but this code isn't meant to deal with `base` yet 😅 see the tests. Do you want me to deal with `base` in this PR?

Reply:
If the code here is meant as a quick prototype of the feature, then I can definitely close one eye here and have it polished later 😄 Just pointing out a bug that could happen here since I'm not sure in further reviews I'll see this diff again.

I don't mind if you want to skip this for now, and fix it later though. But I think it should be fixed after things are settled.

Reply:
Added the checks